### PR TITLE
Cr 1258 | logging improvements and dependencies version update

### PR DESCRIPTION
**Pull request by akritisinhaa**
1. Updating logging to be helpful for creation dashboard in GCP
2. Updating dependencies version

### CODE REVIEW

**Review by devenrawat — APPROVED** [review]

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java [hunk 1]
```diff
@@ -48,7 +48,7 @@
     String methodName = "createNewCase";
     log.with("UPRN", requestBodyDTO.getUprn())
         .with("requestBody", requestBodyDTO)
-        .info("Entering POST {}", methodName);
+        .info("Entering POST for createNewCase");
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java:51]: why? what was wrong with this prev? we now have a string literal created for single use?

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java [hunk 2]
```diff
@@ -77,8 +77,8 @@ private void validateMatchingCaseId(UUID caseId, UUID dtoCaseId, String dtoName)
       throws CTPException {
     if (!caseId.equals(dtoCaseId)) {
       String message =
-          "The caseid in the " + dtoName + " URL does not match the caseid in the request body";
-      log.with(caseId).warn(message);
+          "The caseid in the given dto URL does not match the caseid in the request body";
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java:80]: "The path parameter caseId does not match the caseId in the request body"

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java [hunk 3]
```diff
@@ -99,7 +99,7 @@ private void validateMatchingCaseId(UUID caseId, UUID dtoCaseId, String dtoName)
     String methodName = "modifyAddress";
     log.with("pathParam.caseId", caseId)
         .with("requestBody", addressChange)
-        .info("Entering PUT {}", methodName);
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java:102]: Ditto above - we have methodName

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java [hunk 4]
```diff
@@ -125,7 +125,7 @@ public void fulfilmentRequestBySMS(
     String methodName = "fulfilmentRequestBySMS";
     log.with("pathParam.caseId", caseId)
         .with("requestBody", requestBodyDTO)
-        .info("Entering POST {}", methodName);
+        .info("Entering POST for fulfilmentRequestBySMS");
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java:128]: ditto ditto

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java [hunk 5]
```diff
@@ -139,7 +139,7 @@ public void fulfilmentRequestByPost(
       @Valid @RequestBody PostalFulfilmentRequestDTO requestBodyDTO)
       throws CTPException {
     String methodName = "fulfilmentRequestByPost";
-    log.with("pathParam.caseId", caseId).info("Entering POST {}", methodName);
+    log.with("pathParam.caseId", caseId).info("Entering POST for fulfilmentRequestByPost");
```
Comment (philwhiles) [inline src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java:142]: !

File: src/test/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpointUnitTest.java [hunk 6]
```diff
@@ -53,7 +53,7 @@
   private static final String CASEID_UPRN_INCONSISTENT =
       "The UPRN of the referenced Case and the provided Address UPRN must be matching";
   private static final String CASEID_INCONSISTENT =
-      "The caseid in the modifyAddress URL does not match the caseid in the request body";
+      "The caseid in the given dto URL does not match the caseid in the request body";
```
Comment (philwhiles) [inline src/test/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpointUnitTest.java:56]: See above

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java (hunk 1 above)
Comment (akritisinhaa) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java]: done

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java (hunk 2 above)
Comment (akritisinhaa) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java]: done

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java (hunk 3 above)
Comment (akritisinhaa) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java]: done

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java (hunk 4 above)
Comment (akritisinhaa) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java]: done

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java (hunk 5 above)
Comment (akritisinhaa) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java]: done

File: src/test/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpointUnitTest.java (hunk 6 above)
Comment (akritisinhaa) [reply src/test/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpointUnitTest.java]: done

File: src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java (hunk 2 above)
Comment (philwhiles) [reply src/main/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpoint.java]: "The path parameter caseId does not match the caseId in the request body" is clearer.

File: src/test/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpointUnitTest.java (hunk 6 above)
Comment (philwhiles) [reply src/test/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpointUnitTest.java]: "The path parameter caseId does not match the caseId in the request body" is clearer

**Review by philwhiles — APPROVED** [review]

**Pull request merged**